### PR TITLE
Strip www, m from source_domains

### PR DESCRIPTION
**Pull request by lmcnulty**
Fixes item 1 of #630. Strips www and m from source_domains on submission and adds database migration doing so for existing items.

### CODE REVIEW

**Review by cesarvarela — COMMENTED** [review]
Good catch here! Let's wait to merge this after https://github.com/responsible-ai-collaborative/aiid/pull/685 because I've found some inconsistencies with this field there too.

Please add/update related tests 🙏

**Comment by lmcnulty** [comment]
Again, the "[Generate a Algolia index](https://app.netlify.com/sites/staging-aiid/deploys/62a355c9d49f72000853af12#L623)" test is failing, and I'm pretty sure it has to do with the schema changes from #685.

File: site/gatsby-site/src/components/forms/QuickAddForm.js [hunk 1]
```diff
@@ -44,7 +44,7 @@ const QuickAddForm = ({ className = '' }) => {
         const quickAdd = {
           incident_id: '0',
           date_submitted: format(new Date(), 'yyyy-MM-dd'),
-          source_domain: url.hostname,
+          source_domain: url.hostname.replace(/^(www|m)\./, ''),
```
[the remark this answers is not in the document]
Comment (lmcnulty) [reply site/gatsby-site/src/components/forms/QuickAddForm.js]: I've extracted it to `getSourceDomain` which is in its own file in `util`.

File: site/gatsby-site/migrations/2022.06.07T17.24.11.normalize-domains.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      batch = [];
+    }
+  }
+};
+
+/**
+ * @type {import('umzug').MigrationFn<any>}
+ * @param {{context: {client: import('mongodb').MongoClient}}} context
+ */
+exports.down = async ({ context: { client } }) => {
```
[the remark this answers is not in the document]
Comment (cesarvarela) [reply site/gatsby-site/migrations/2022.06.07T17.24.11.normalize-domains.js:63]: Line 65, the `down` operation, doesn't undo the `up` operation. It seems unrelated.

[the remark this answers is not in the document]
Comment (lmcnulty) [reply site/gatsby-site/migrations/2022.06.07T17.24.11.normalize-domains.js]: Oh, whoops. Thanks!

**Comment by cesarvarela** [comment]
I'm looking into why the test is failing

**Comment by cesarvarela** [comment]
Fixed here https://github.com/responsible-ai-collaborative/aiid/pull/699

File: site/gatsby-site/cypress/integration/submit.js [hunk 3]
```diff
@@ -15,7 +15,7 @@ describe('The Submit form', () => {
     cy.intercept('GET', parserURL, parseNews).as('parseNews');
 
     cy.get('input[name="url"]').type(
-      `https://arstechnica.com/gadgets/2017/11/youtube-to-crack-down-on-inappropriate-content-masked-as-kids-cartoons/`
+      `https://www.arstechnica.com/gadgets/2017/11/youtube-to-crack-down-on-inappropriate-content-masked-as-kids-cartoons/`
```
Comment (smcgregor) [inline site/gatsby-site/cypress/integration/submit.js:18]: Why was it necessary to add `www` to these? Shouldn't it still work with this change?

**Comment by smcgregor** [comment]
One question, but otherwise 👍 

@cesarvarela can you take a final review?

**Review by smcgregor — APPROVED** [review]

**Review by cesarvarela — APPROVED** [review]

**Pull request merged**